mllib/src/main/scala/org/apache/spark/ml/classification/Classifier.scala

```diff
@@ -115,7 +116,16 @@ abstract class ClassificationModel[FeaturesType, M <: ClassificationModel[Featur
         }
         predictUDF(col(getFeaturesCol))
       }
-      outputData = outputData.withColumn(getPredictionCol, predUDF)
+      // determine number of classes either from metadata if provided.
+      val labelSchema = dataset.schema($(labelCol))
+      // extract label metadata from label column if present, or create a nominal attribute
+      // to output the number of labels
+      val labelAttribute = Attribute.fromStructField(labelSchema) match {
```

Comment:
1.  This is why we try to keep the constructors private for as long as possible.  : )  But we could get around it by creating a ```private[ml]``` setter method which can set the labelAttribute after the model is created.
2. You're right; I think that's a bug technically.  Here's a JIRA: https://issues.apache.org/jira/browse/SPARK-14926